### DISCUSSION TITLE
Enable Image Scanning via Input File Path for TAR File Formats

### DISCUSSION THREAD

**Discussion by KiranWadeyar** [fatihtokus/scan2html#68] [Ideas]
**Description:**
This feature request aims to enhance the image scanning functionality by allowing users to specify an input file path instead of an image name. This enhancement will significantly benefit scenarios involving TAR file formats, providing a more flexible and streamlined workflow for image scanning processes.

**Current Limitation:**
Currently, the system requires an image name for scanning, which is not suitable for TAR file formats.

**Proposed Solution:**
Implement support for scanning images using a specified file path, including support for TAR file formats. Trivy already supports it with --input tag

**Benefits:**
Improved flexibility and usability for users dealing with various image formats.
Streamlined integration with existing workflows that involve TAR files.
Enhanced efficiency in image scanning processes by eliminating the need to convert TAR files to other formats.

**Comment by fatihtokus** [comment]
Hi @KiranWadeyar ,

Thank you for your interest in the plugin. The good news is that if that feature is supported by trivy then it is supported by scan2html as well :)
I have tested it out for you:
1.   docker pull ruby:3.1-alpine3.15
2.   docker save ruby:3.1-alpine3.15 -o ruby-3.1.tar
3.  trivy scan2html image --input ruby-3.1.tar interactive_report_tar.html
[interactive_report_tar.html.zip](https://github.com/user-attachments/files/16177928/interactive_report_tar.html.zip)

Please try and let us know.
Please consider giving us a GitHub star ⭐️. Thank you!

Regards,
Fatih

**Comment by fatihtokus** [comment]
Hi @KiranWadeyar ,

Have you had a change to look at this?
